src/lib/molecules/TimePicker/TimePicker.tsx

```diff
@@ -34,49 +39,127 @@ function generateTimeValues(format, max) {
+function convertToFormat(value: string, format: string | string[] | undefined, notEmpty: boolean) {
     if (!value && notEmpty) {
-        return format.length === 1 ? '0' : '00';
+        return format?.length === 1 ? '0' : '00';
     }
-    if (format.length === 2 && value.length === 1) {
+    if (format?.length === 2 && value?.length === 1) {
         return `0${value}`;
     }
-    if (format.length === 1 && value.length === 2) {
+    if (format?.length === 1 && value?.length === 2) {
         return Number(value).toString();
     }
 
     return value;
 }
 
-function TimePicker({
+const TimePickerConfig = {
+    appearance: ['multipleInputs', 'singleInput'],
+    screenTypes: ['desktop', 'mobile'],
+    positions: ['top', 'right', 'bottom', 'left']
+} as const;
+
+export type GetArrayAsUnion<T extends keyof typeof TimePickerConfig> = (typeof TimePickerConfig)[T][number];
+
+interface ITimePickerProps {
+    /**
+     * Define is seconds field will shown or no
+     */
+    showSeconds?: boolean;
+    /**
+     * Select view with multiple inputs or with single inputs
+     */
+    appearance?: GetArrayAsUnion<'appearance'>;
+    /**
+     * Format for hour field
+     */
+    hourFormat?: 'HH' | 'H' | 'hh' | 'h';
+    /**
+     * Format for hour field
+     */
+    minuteFormat?: 'mm' | 'm';
+    /**
+     * Format for hour field
+     */
+    secondFormat?: 'ss' | 's';
+    /**
+     * Time field separator
+     */
+    separator?: string;
+    /**
+     * Fires an event when field is changed((event: SyntheticEvent) => void).
+     */
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    /**
+     * Fires an event when field is blurred((date: Date, event: SyntheticEvent) => void).
+     */
+    onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
+    /**
+     * Additional classname
+     */
+    className?: string;
+    /**
+     * Value for input field
+     */
+    value?: string;
+    /**
+     * disabled for input field
+     */
+    disabled?: boolean;
+    /**
+     * Makes Time picker readonly when set to "true"
+     */
+    readOnly?: boolean;
+    /**
+     * The switch between mobile and desktop version of Dropdown will be applied automatically, when the prop is not specified.
+     * When the prop is present it must be changed from outside.
+     */
+    screenType?: GetArrayAsUnion<'screenTypes'>;
+    /**
+     * preferred positions by priority ['bottom','top', 'left', 'right']
+     * default is ['bottom','top', 'left', 'right']
+     * if you'd like, you can limit the positions ['top', 'left']
+     */
+    positions: GetArrayAsUnion<'positions'>;
+}
+
+export interface ChildRef {
+    toggleOpen: () => void;
+}
+
+const TimePicker: React.FC<ITimePickerProps> = ({
     value,
-    onChange,
-    showSeconds,
-    appearance,
-    hourFormat,
-    minuteFormat,
-    secondFormat,
-    separator,
+    onChange = noop,
+    showSeconds = true,
+    appearance = timePickerConfig.appearance[0],
```

Comment:
Please change to have ridable documentation in storybook.
```suggestion
    appearance = 'multipleInputs',
```